db/keyval/filedb/decoder/decoder_json.go

```diff
@@ -0,0 +1,88 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package decoder
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Extension supported by this decoder
+var jsonDcExtensions = []string{".json"}
+
+// JSONDecoder can be used to decode json-type files
+type JSONDecoder struct{}
+
+// Represents data structure of json file used for configuration
+type jsonFile struct {
+	Data []jsonFileEntry `json:"data"`
+}
+
+// Single record of key-value, where key is defined as string, and value is modeled as raw message
+// (rest of the json file under the "value").
+type jsonFileEntry struct {
+	Key   string          `json:"key"`
+	Value json.RawMessage `json:"value"`
+}
+
+// NewJSONDecoder creates a new decoder instance
+func NewJSONDecoder() *JSONDecoder {
+	return &JSONDecoder{}
+}
+
+// IsProcessable returns true if decoder is able to decode provided file
+func (jd *JSONDecoder) IsProcessable(file string) bool {
+	for _, ext := range jsonDcExtensions {
+		if strings.HasSuffix(file, ext) {
+			return true
+		}
+	}
+	return false
+}
+
+// Encode provided file entries into JSON byte set
+func (jd *JSONDecoder) Encode(data []*FileDataEntry) ([]byte, error) {
+	// Convert to json-specific structure
+	var jsonFileEntries []jsonFileEntry
+	for _, dataEntry := range data {
+		jsonFileEntries = append(jsonFileEntries, jsonFileEntry{
+			Key:   dataEntry.Key,
+			Value: dataEntry.Value,
+		})
+	}
+	// Encode to type specific structure
+	jsonFile := &jsonFile{Data: jsonFileEntries}
+	return json.Marshal(jsonFile)
+}
+
+// Decode provided byte set of json file and returns set of file data entries
+func (jd *JSONDecoder) Decode(byteSet []byte) ([]*FileDataEntry, error) {
+	if byteSet == nil || len(byteSet) == 0 {
```

Comment:
It's enough to check `len(byteSet) == 0`, since it will evaluate to true for `nil`.